### ISSUE TITLE
NP_ALWAYS_NULL_EXCEPTION does not recognize Objects.nonNull as an acceptable null guard

### ISSUE THREAD

**Issue by unfuse** [spotbugs/spotbugs#1991]
Cleaning up some SpotBugs lint for my project, and came across a couple NPE warnings (excellent!) 

Specifically it found an exception each of two varieties: `NP_NULL_ON_SOME_PATH_EXCEPTION` and `NP_ALWAYS_NULL_EXCEPTION`. I have included redacted source code as examples

For `NP_NULL_ON_SOME_PATH_EXCEPTION` I was able to add `Objects.nonNull` (our preferred null guard) to fix it:

```java
        File file = null;
        try {
            file = File.createTempFile(SUFFIX_1, SUFFIX_2);
            this.foo.write(file.getAbsolutePath());
            // utility that reads file to a string and returns it. we didn't do anything particularly fancy here
            // throws IOException
            return FileUtils.readText(file);
        }
        catch (final IOException ex) {
            // SpotBugs found this NP_NULL_ON_SOME_PATH_EXCEPTION for us
            if (Objects.nonNull(file)) {
                LOG.info(String.format("Exception encountered while writing output to a file: %s",
                        file.getAbsolutePath()), ex);
            }
            return StringUtils.EMPTY;
        }
        finally {
            // Noting that Objects.nonNull is OK here too. SpotBugs does not complain about this.
            if (Objects.nonNull(file)) {
                IOUtils.delete(file.getAbsolutePath());
            }
        }
```

But I could not do the same for `NP_ALWAYS_NULL_EXCEPTION`

```java
        Object result = null;
        try {
            result = doThingThatCanThrowException();
        }
        catch (final Exception e) {
            // Log data and exception and capture a "default" 
            result = handleFailureAndCreateOutput(e);
        }
        finally {
            // Cannot use Objects.nonNull here, it will claim NP_ALWAYS_NULL_EXCEPTION anyway. result != null works. 
            // if (Objects.nonNull(result)) {
            if (result != null) {
                // SpotBugs identifies this specific line as the problem! 
                Foo.use(result.isFailure());
            }
        }
        return result;
```

The cases seem strikingly similar but I don't understand the particular details why it is deciding an ALWAYS_NULL vs not.

It seems inconsistent that `Objects.nonNull` works for some lints but not others. Would be nice if this could be addressed if its simple!

**Comment by giovanni-nabil** [comment]
same issue with bug type `UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR` no respect for `Objects.nonNull()` nor `Optional.ofNullable()`
